packages/integration-aws-node/src/DefaultAwsCredentialsManager.test.ts

```diff
@@ -24,12 +24,20 @@ import {
 } from '@aws-sdk/client-sts';
 import { Config, ConfigReader } from '@backstage/config';
 import { promises } from 'fs';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
 
 const env = process.env;
 let stsMock: AwsClientStub<STSClient>;
 let config: Config;
 
 jest.mock('fs', () => ({ promises: { readFile: jest.fn() } }));
+jest.mock('@aws-sdk/credential-providers', () => {
+  const originalModule = jest.requireActual('@aws-sdk/credential-providers');
+  return {
+    ...originalModule,
+    fromNodeProviderChain: jest.fn(),
```

Comment:
probably with this change you won't need the `.mockImplementation()` block below

```suggestion
    fromNodeProviderChain: jest.fn(originalModule.fromNodeProviderChain),
```

Reply:
The `mockImplementation` block is necessary due to the `jest.resetAllMocks()` call within the `beforeEach`. Otherwise, yes I'd be able to remove it.